### ISSUE TITLE
Complex Normal Random Variable

### ISSUE THREAD

**Issue by Rassibassi** [blei-lab/edward#742]
Hej,
I've been playing around with Edward and think I will use it for my research or try to. I am missing the complex Normal Random Variable. As far as I can tell there is no support for that.
In `random_variables.py`:
```
...
Normal.support = 'real'
...
```
How and where would I start to make this also support complex?

Thanks for the great work!

**Comment by dustinvtran** [comment]
I'm not familiar with complex random variables to answer this unfortunately. Do you have a good reference and/or application in mind?

**Comment by Rassibassi** [comment]
Yes, I am a communication/information theory PhD student and most of the time we are working with complex signals (deterministic or random/noisy). I have been looking into the AR(p) process examples from here https://github.com/blei-lab/edward/issues/211 and here https://github.com/nfoti/edward/tree/wip-ts-examples/examples/time-series

With your example from the above issue, I can explain what I want to do:
```
mu = Normal(mu=1.0, sigma=10.0)
beta = Normal(mu=1.0, sigma=10.0)

noise_proc = InverseGamma(alpha=1.0, beta=1.0)
noise_obs = InverseGamma(alpha=1.0, beta=1.0)

x = [0] * N
x[0] = Normal(mu=mu, sigma=10.0)  # fat prior on x
for n in range(1, N):
  x[n] = Normal(mu=mu + beta * x[n-1], sigma=noise_proc)

y = Normal(mu=x, sigma=noise_obs)
```
if we look at this from a filtering perspective, we filter a real Gaussian Noise source with the filter coefficients `beta` and obtain a noise signal `x[n]`. All real.

If we now choose a complex Gaussian Noise source and filter with the same coefficients, we would obtain a complex Gaussian Noise source.
See https://en.wikipedia.org/wiki/Complex_normal_distribution
tl;dr: 
The complex Normal Distribution is basically a two dimensional Gaussian distribution (looks a bit like a Sombrero ;) ), and the two dimensions refer to real and imag part of a complex random variable. There is a "standard complex normal" and "circularly-symmetric complex normal", but let's not get confused by that and stick with the "standard complex normal". "standard complex normal" means, in the two dimensional vector representation, that the two dimensions are independent with equal variance (scaled identity matrix as co-variance matrix).

I'd love to have something like this:
```
x = [0] * N
x[0] = ComplexNormal(mu=cmplxMu, sigma=covarScalar)  # fat prior on x
for n in range(1, N):
  x[n] = ComplexNormal(mu=cmplxMu+ beta * x[n-1], sigma=noise_proc)
```
where `cmplxMu` is the complex mean, `covarScalar` the scalar of the co-variance matrix and thus `x[n]` will be complex, too.

I can solve this by just creating two real Gaussian Noise sources (one for the real part and one for the imaginary part). Then, I can write all my equations separated by real and imag part (in a vector representation).  Unfortunately, it gets rather messy for high order AR(p) and I would favour to implement support for complex numbers here.

**Comment by dustinvtran** [comment]
Got it. Have you looked at http://edwardlib.org/api/model-development yet? Seems like you can use that and build off the [`normal.py`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/distributions/normal.py) source code using `tf.complex`.
